python/tank/hook.py

```diff
@@ -206,6 +208,24 @@ def parent(self):
         """
         return self.__parent
 
+    def get_instance(self):
```

Comment:
I expected a `bundle.get_hook_instance()` or `bundle.create_hook_instance()`.

On a more general note, since nothing you've done here involved modifying stuff outside of the hook class, couldn't we have added the `get_instance` method in the app itself for the time being? Are you seeing other uses for this outside of the publier tool at the moment? Maybe its one of these things we could have kept in an app for now and expanded upon in core in the future when the need arises.